assignment-03/app/Dao/StudentDao.php

```diff
@@ -0,0 +1,83 @@
+<?php
+
+namespace App\Dao;
+
+use App\Contracts\Dao\StudentDaoInterface;
+use App\Models\Student;
+use App\Models\Major;
+
+class StudentDao implements StudentDaoInterface
+{
+    /**
+     * sample function
+     *
+     * @return string
+     */
+    public function getAllStudents()
+    {
+        return Student::select('students.*')
+                            ->leftJoin('majors','students.major_id','majors.id')
+                            ->when(request('searchKey'),function($query){
+                            $key=request('searchKey');
+                            $query->where('students.name', 'like', '%' . $key . '%')
+                            ->orWhere('students.email', '=', $key)
+                            ->orWhere('students.phone', '=', $key)
+                            ->orWhere('students.address', 'like', '%' . $key . '%')
+                            ->orWhere('majors.name', 'like', '%' . $key . '%');
+                            })->get();
```

Comment:
- Please make vertically align `:: & ->`
- Fix query indentation.
- Search major name with where = . Not where like.
- How about raw query?